test/box-luatest/gh_7954_sort_spaces_in_snap_by_id_test.lua

```diff
@@ -0,0 +1,46 @@
+local t = require('luatest')
+local g = t.group('gh-7954')
+
+g.before_all(function(cg)
+    local server = require('luatest.server')
+    cg.server = server:new({alias = 'master'})
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+-- Check that spaces in the snapshot file are strictly ordered.
+g.test_spaces_in_snap = function(cg)
+    local fio = require('fio')
+    local xlog = require('xlog')
+
+    -- Create some spaces, fill them with data, and write a snapshot.
+    local snap_name = cg.server:exec(function()
+        local spaces = {}
+        for i, id in ipairs{739, 761, 863, 881, 907, 937, 1009, 1033, 1039} do
```

Comment:
This test doesn't check that spaces with id < 256 are stored after system spaces.

Reply:
Added spaces with id < 256